source/scripts/Background/controllers/MessageHandler/requests.ts

```diff
@@ -1,276 +1,307 @@
+      const decodedTxData = data?.data ? decoder.decodeData(data?.data) : null;
+
+      const sendTransaction = async () => {
+        await masterController.createPopup(windowId, message.data.network, 'sendTransaction', { ...data });
+
+        setPendingWindow(true);
+
+        window.addEventListener(
+          'transactionSent',
+          (ev: any) => {
+            console.log('Connect window addEventListener', ev.detail);
+            if (ev.detail.windowId === windowId) {
+              port.postMessage({
+                id: message.id,
+                data: { result: ev.detail.result, data: {} },
+              });
+              setPendingWindow(false);
             }
+          },
+          { once: true, passive: true }
+        );
+      };
 
-            if (decodedTxData?.method === 'approve') {
-                return approveSpend();
-            } else {
-                return sendTransaction();
+      const approveSpend = async () => {
+        //special case transaction
+        await masterController.createPopup(windowId, message.data.network, 'approveSpend', { ...data });
+
+        setPendingWindow(true);
+
+        window.addEventListener(
+          'spendApproved',
+          (ev: any) => {
+            console.log('Connect window addEventListener', ev.detail);
+            if (ev.detail.windowId === windowId) {
+              port.postMessage({
+                id: message.id,
+                data: { result: ev.detail.result, data: {} },
+              });
+              setPendingWindow(false); //prevents popup?
             }
+          },
+          { once: true, passive: true }
+        );
+      };
 
-        // } else if (method === 'wallet.setLedgerAccounts') {
-        //     await window.controller.stargazerProvider.importLedgerAccounts(args[0]);
-        //     // port.postMessage({ id: message.id, data: { result: "success" } });
-        //   return Promise.resolve({ id: message.id, result: "success" });
-        // } else if (method === 'wallet.postTransactionResult') {
-        //   await window.controller.stargazerProvider.postTransactionResult(args[0]);
-        //   return Promise.resolve({ id: message.id, result: "success" });
-    }
+      if (decodedTxData?.method === 'approve') {
+        return approveSpend();
+      } else {
+        return sendTransaction();
+      }
 
-    if (result !== undefined) {
-        return Promise.resolve({id: message.id, result});
-    }
+    // } else if (method === 'wallet.setLedgerAccounts') {
+    //     await window.controller.stargazerProvider.importLedgerAccounts(args[0]);
+    //     // port.postMessage({ id: message.id, data: { result: "success" } });
+    //   return Promise.resolve({ id: message.id, result: "success" });
+    // } else if (method === 'wallet.postTransactionResult') {
+    //   await window.controller.stargazerProvider.postTransactionResult(args[0]);
+    //   return Promise.resolve({ id: message.id, result: "success" });
```

Comment:
We can get rid of this commented out code

Reply:
@cmirandac665 is this necessary for your work with ledger?